astroquery/alma/core.py

```diff
@@ -210,44 +216,61 @@ def __init__(self):
         self._sia = None
         self._tap = None
         self._datalink = None
-        self.sia_url = None
-        self.tap_url = None
-        self.datalink_url = None
+        self._sia_url = None
+        self._tap_url = None
+        self._datalink_url = None
 
     @property
     def datalink(self):
         if not self._datalink:
-            base_url = self._get_dataarchive_url()
-            if base_url.endswith('/'):
-                self.datalink_url = base_url + ALMA_DATALINK_PATH
-            else:
-                self.datalink_url = base_url + '/' + ALMA_DATALINK_PATH
-            self._datalink = pyvo.dal.adhoc.DatalinkService(
-                baseurl=self.datalink_url)
+            self._datalink = pyvo.dal.adhoc.DatalinkService(self.datalink_url)
         return self._datalink
 
+    @property
+    def datalink_url(self):
+        if not self._datalink_url:
+            try:
+                self._datalink_url = f"{self._get_dataarchive_url()}{DATALINK_SERVICE_PATH}"
+            except requests.exceptions.HTTPError as err:
+                log.debug(
+                    f"ERROR getting the CADC registry: {str(err)}")
```

Comment:
Here and below, too

```suggestion
                    f"ERROR getting the ALMA registry: {str(err)}")
```

Reply:
though maybe the whole message needs an update as at the end, if I see it correctly, you don't use the registry.

Reply:
The entire message was inappropriate.  It has been modified.